examples/tv-casting-app/linux/main.cpp

```diff
@@ -237,23 +227,280 @@ void OnContentLauncherFailureResponse(void * context, CHIP_ERROR error)
+    }
+    gTargetVideoPlayerInfo.PrintInfo();
+}
+
+void OnDescriptorReadFailureResponse(void * context, CHIP_ERROR error)
+{
+    ChipLogError(AppServer, "Descriptor: Default Failure Response: %" CHIP_ERROR_FORMAT, error.Format());
+}
+
+void ReadServerClusters(EndpointId endpointId)
+{
+    Server * server           = &(chip::Server::GetInstance());
+    chip::FabricInfo * fabric = server->GetFabricTable().FindFabricWithIndex(gTargetVideoPlayerInfo.GetFabricIndex());
+    if (fabric == nullptr)
+    {
+        ChipLogError(AppServer, "Did not find fabric for index %d", gTargetVideoPlayerInfo.GetFabricIndex());
+        return;
+    }
+
+    chip::DeviceProxyInitParams initParams = {
+        .sessionManager = &(server->GetSecureSessionManager()),
+        .exchangeMgr    = &(server->GetExchangeManager()),
+        .idAllocator    = &(server->GetSessionIDAllocator()),
+        .fabricTable    = &(server->GetFabricTable()),
+        .clientPool     = &gCASEClientPool,
+    };
+
+    PeerId peerID                                         = fabric->GetPeerIdForNode(gTargetVideoPlayerInfo.GetNodeId());
+    chip::OperationalDeviceProxy * operationalDeviceProxy = chip::Platform::New<chip::OperationalDeviceProxy>(initParams, peerID);
+    if (operationalDeviceProxy == nullptr)
+    {
+        ChipLogError(AppServer, "Failed in creating an instance of OperationalDeviceProxy");
+        return;
+    }
+
+    SessionHandle handle = server->GetSecureSessionManager().FindSecureSessionForNode(gTargetVideoPlayerInfo.GetNodeId());
+    operationalDeviceProxy->SetConnectedSession(handle);
+
+    chip::Controller::DescriptorCluster cluster;
+    CHIP_ERROR err = cluster.Associate(operationalDeviceProxy, endpointId);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "Associate() failed: %" CHIP_ERROR_FORMAT, err.Format());
+        return;
+    }
+
+    TargetEndpointInfo * endpointInfo = gTargetVideoPlayerInfo.GetAddEndpoint(endpointId);
+
+    if (cluster.ReadAttribute<app::Clusters::Descriptor::Attributes::ServerList::TypeInfo>(
+            endpointInfo, OnDescriptorReadSuccessResponse, OnDescriptorReadFailureResponse) != CHIP_NO_ERROR)
+    {
+        ChipLogError(Controller, "Could not read Descriptor cluster ServerList");
+    }
+
+    ChipLogProgress(Controller, "Sent descriptor read for remote endpoint=%d", endpointId);
+}
+
+void ReadServerClustersForNode(NodeId nodeId)
+{
+    for (const auto & binding : BindingTable::GetInstance())
+    {
+        ChipLogProgress(NotSpecified,
+                        "Binding type=%d fab=%d nodeId=0x" ChipLogFormatX64
+                        " groupId=%d local endpoint=%d remote endpoint=%d cluster=" ChipLogFormatMEI,
+                        binding.type, binding.fabricIndex, ChipLogValueX64(binding.nodeId), binding.groupId, binding.local,
+                        binding.remote, ChipLogValueMEI(binding.clusterId.ValueOr(0)));
+        if (nodeId == binding.nodeId)
```

Comment:
This is not going to behave correctly (or even sanely) for groupcast bindings....